File: src/main/java/no/nav/syfo/web/selftest/SelftestController.java
package no.nav.syfo.web.selftest;

import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.jdbc.core.JdbcTemplate;

@Slf4j
@RestController
@RequestMapping(value = "/internal")
public class SelftestController {
    private static final String APPLICATION_LIVENESS = "Application is alive!";
    private static final String APPLICATION_READY = "Application is ready!";

    @Autowired
    private JdbcTemplate template;

    @ResponseBody
    @RequestMapping(value = "/isAlive", produces = MediaType.TEXT_PLAIN_VALUE)
    public String isAlive() {
        checkDatabase();
        return APPLICATION_LIVENESS;
    }

    @ResponseBody
    @RequestMapping(value = "/isReady", produces = MediaType.TEXT_PLAIN_VALUE)
    public String isReady() {
        checkDatabase();
        return APPLICATION_READY;
    }

    private void checkDatabase(){
        template.execute("select now();");
        log.error("Klarte å koble mot databasen");
    }
}

File: src/main/kotlin/no/nav/syfo/web/selftest/SelftestController.kt
package no.nav.syfo.web.selftest
/*

import org.springframework.beans.factory.annotation.Autowired

@Slf4j
@RestController
@RequestMapping(value = "/internal")
class SelftestController {
    @Autowired
    private val template: JdbcTemplate? = null

    @get:RequestMapping(value = "/isAlive", produces = MediaType.TEXT_PLAIN_VALUE)
    @get:ResponseBody
    val isAlive: String
        get() {
            checkDatabase()
            return APPLICATION_LIVENESS
        }

    @get:RequestMapping(value = "/isReady", produces = MediaType.TEXT_PLAIN_VALUE)
    @get:ResponseBody
    val isReady: String
        get() {
            checkDatabase()
            return APPLICATION_READY
        }

    private fun checkDatabase() {
        template.execute("select now();")
    }

    companion object {
        private const val APPLICATION_LIVENESS = "Application is alive!"
        private const val APPLICATION_READY = "Application is ready!"
    }
}
*/
